pkg/vex/csaf.go

```diff
@@ -41,45 +41,102 @@ func (v *CSAF) affected(vuln *csaf.Vulnerability, pkgURL *packageurl.PackageURL)
 		return true
 	}
 
-	var status Status
-	switch {
-	case v.matchPURL(vuln.ProductStatus.KnownNotAffected, pkgURL):
-		status = StatusNotAffected
-	case v.matchPURL(vuln.ProductStatus.Fixed, pkgURL):
-		status = StatusFixed
+	matchProduct := func(purls []*purl.PackageURL, pkgURL *packageurl.PackageURL) bool {
+		for _, p := range purls {
+			if p.Match(pkgURL) {
+				return true
+			}
+		}
+		return false
 	}
 
-	if status != "" {
-		v.logger.Infow("Filtered out the detected vulnerability",
-			zap.String("vulnerability-id", string(*vuln.CVE)),
-			zap.String("status", string(status)))
-		return false
+	for _, product := range lo.FromPtr(vuln.ProductStatus.KnownNotAffected) {
```

Comment:
Is it possible not to repeat almost the same logic?

Reply:
done at https://github.com/aquasecurity/trivy/pull/5923/commits/9cbfc6bd4a4c9505ddbaa9becf27a105e52476a3